src/types/flows.ts

```diff
@@ -141,7 +141,10 @@ export interface VariableMap {
 
 // TODO: type this better. there are required properties, that have types, but
 // we also need this to stay open for panels to register whatever they want
-export interface QueryScope {
+interface VariableOuterScope {
+  vars?: Variable[]
+}
+export interface QueryScope extends VariableOuterScope {
```

Comment:
This use of `[props: string]: any` means that the TimeRange variables (and other variables) are easily overriden with anything. And the dev cannot lean on typescript. 😭

Reply:
This code was left in. Even though it's wrong, per the explanation^^ above.
But I'm leaving notebooks code alone for now, which means leaving this alone.